python/dgl/nn/pytorch/factory.py

```diff
@@ -81,11 +81,28 @@ def forward(self, x, algorithm='topk'):
         algorithm : str, optional
             Algorithm used to compute the k-nearest neighbors.
 
-            * 'topk' will use topk algorithm (quick-select or sorting,
-            depending on backend implementation)
-            * 'kd-tree' will use kd-tree algorithm (cpu version)
+            * 'bruteforce-blas' will first compute the distance matrix
+              using BLAS matrix multiplication operation provided by
+              backend frameworks. Then use topk algorithm to get
+              k-nearest neighbors. This method has the largest memory
+              overhead.
 
-            (default: 'topk')
+            * 'bruteforce' will compute distances pair by pair and
+              directly select the k-nearest neighbors during distance
+              computation. This method is slower than 'bruteforce-blas'
+              but has less memory overhead since we do not need to store
+              all distances.
+
+            * 'bruteforce-sharemem' (CUDA only) is similar to 'bruteforce'
+              but use shared memory in CUDA devices for buffer. This method is
+              faster than 'bruteforce' when the dimension of input points
+              is not large. This method is only available on CUDA device.
```

Comment:
When the feature dimension of points is large, e.g. 4096 on V100 GPU, 'bruteforce-sharemem' is slower since we only have limited shared memory and high-dimensional data will cause low degree of parallelism.

e.g.:
P =   1x  1000, F = 4096, k = 16, bfs_time: 1.6141, bf_time: 1.4211